velox/serializers/PrestoSerializer.h

```diff
@@ -64,11 +82,16 @@ class PrestoVectorSerde : public VectorSerde {
       StreamArena* streamArena,
       const Options* options) override;
 
-  /// Serializes a flat RowVector with possibly encoded children. Preserves
-  /// first level of encodings. Dictionary vectors must not have nulls added by
-  /// the dictionary.
+  /// Serializes a single RowVector with possibly encoded children, preserving
+  /// their encodings. Encodings are preserved recursively for any RowVector
+  /// children, but not for children of other nested vectors such as Array, Map,
+  /// and Dictionary.
   ///
-  /// Used for testing.
+  /// PrestoPage does not support serialization of Dictionaries with nulls;
+  /// in case dictionaries contain null they are serialized as flat buffers.
```

Comment:
Would it be safer to throw in this case? Will users be surprised to later find their vectors flattened?

Reply:
potentially, but this will change the API contract and likely break downstream consumers of this API. I think this behavior will be easier to change as we refactor this code.